### PR TITLE
chore: remove rdf-js which is deprecated

### PR DESCRIPTION
**Pull request by tpluscode**
It is high time we remove the [deprecated `rdf-js`](https://www.npmjs.com/package/rdf-js) package and directly use `@rdfjs/types`. I also update the `notNeededPackages.json` to reflect that.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/rdfjs__serializer-jsonld/index.d.ts
1. types/rdfjs__serializer-jsonld/rdfjs__serializer-jsonld-tests.ts
1. types/rdfjs__formats-common/index.d.ts
1. types/rdfjs__formats-common/rdfjs__formats-common-tests.ts
1. types/rdfjs__parser-jsonld/rdfjs__parser-jsonld-tests.ts

<details>
<summary>as well as these 16 other files...</summary>
<p>types/rdfjs__parser-jsonld/index.d.ts, types/rdfjs__serializer-jsonld-ext/rdfjs__serializer-jsonld-ext-tests.ts, types/rdfjs__serializer-jsonld-ext/index.d.ts, types/rdfjs__express-handler/index.d.ts, types/rdfjs__express-handler/rdfjs__express-handler-tests.ts, types/rdfjs__parser-n3/index.d.ts, types/rdfjs__parser-n3/rdfjs__parser-n3-tests.ts, types/rdf-dataset-indexed/rdf-dataset-indexed-tests.ts, types/rdf-data-model/rdf-data-model-tests.ts, types/rdfjs__serializer-ntriples/index.d.ts, types/rdfjs__fetch/rdfjs__fetch-tests.ts, types/rdf-transform-triple-to-quad/rdf-transform-triple-to-quad-tests.ts, types/clownface/clownface-tests.ts, types/rdfjs__serializer-rdfjs/index.d.ts, types/rdfjs__serializer-rdfjs/rdfjs__serializer-rdfjs-tests.ts, types/rdfjs__serializer-ntriples/rdfjs__serializer-ntriples-tests.ts</p>
</details>


Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 376c5891237e6507ffd7a9bda8b83dc0a4bf9341
</p>

File: notNeededPackages.json
Comment (peterblazejewicz) [inline notNeededPackages.json:4580]: not sure about this change. `@rdfjs/types` was never on DT.
The package change happened back in 2020:
https://github.com/rdfjs/types/commit/f97793d9e1ff42ad4113c687dd6996633499eb73
Should it stay as is?

**Comment by tpluscode** [comment]
The truth is, `rdf-js` never really existed as a package in the first place. We created it as compatibility proxy so that it could transparently replace the old types package when it got removed. All the types were moved and are now managed as ***@***.***/types`.

In hindsight, I should have removed all trace of `rdf-js` back in 2020 indeed.

> @rdfjs/types was never on DT. 

Besides, does it really matter? 

Sent from Nine

________________________________

**Comment by peterblazejewicz** [comment]
> Besides, does it really matter?

Yes, the file was edited manually. Let if hangs till week start (so tmrw)

**Comment by tpluscode** [comment]
> > Besides, does it really matter?
> 
> Yes, the file was edited manually. Let if hangs till week start (so tmrw)

I mean, who else would have edited it, other than me manually? 😅 

I can revert this but I thought it makes sense to put the package `rdf-js` behind if it's not at all used anywhere in DT anymore and we recommend installing `@rdfjs/types` in projects

File: notNeededPackages.json
Comment (jakebailey) [reply notNeededPackages.json:4580]: Yes, this file is used specifically for pushing out shim packages and deprecations; this change isn't correct.

**Review by jakebailey — CHANGES_REQUESTED** [review]
Does the real package not export these types? Why go through a side package?

(Marking as needs review for the not needed package needing to be reverted.)

**Comment by jakebailey** [comment]
I can see now that the `rdf-js` package is just:

```ts
export * from '@rdfjs/types'
```

And has been since v2.0.0. So, theoretically this seems alright?

**Review by jakebailey — APPROVED** [review]
Yeah, all packages being replaced are `rdf-js@^4.0.2`.

**Review by peterblazejewicz — APPROVED** [review]
@tpluscode thanks for the patience

**Comment by tpluscode** [comment]
I reverted that change in `notNeededPackages.json`. I don't mind either way, so your call @jakebailey

**Comment by jakebailey** [comment]
I already approved it, unless I'm missing something?

**Comment by tpluscode** [comment]
You approved after commit bf83484
But from your later comments I get the impression that you're fine with my initial proposed changes

**Comment by jakebailey** [comment]
Hm, I don't see any commits after I approved it. Is there something unpushed?

image

**Comment by tpluscode** [comment]
No matter.

Ready to merge

**Comment by tpluscode** [comment]
Thank you @jakebailey @peterblazejewicz

**Pull request merged**
